apps/smart-contracts/token/test/MiniSales.test.ts

```diff
@@ -671,5 +671,105 @@ describe('=> MiniSales', () => {
         })
       })
     })
+
+    describe('if user purchases and is not allowed source', () => {
+      beforeEach(async () => {
+        await sourceAllowlist.connect(owner).set([disallowedSourceUser.address], [false])
+        expect(await sourceAllowlist.isIncluded(disallowedSourceUser.address)).to.eq(false)
+        const disallowedSourceUserBalanceBefore = await ppo.balanceOf(disallowedSourceUser.address)
+        const miniSalesBalanceBefore = await ppo.balanceOf(miniSales.address)
+        // Purchase to disallowed source user
+        await miniSales
+          .connect(purchaser)
+          .purchase(disallowedSourceUser.address, saleTokenAmount, testPrice)
+        expect(await ppo.balanceOf(disallowedSourceUser.address)).to.eq(
+          disallowedSourceUserBalanceBefore.add(saleTokenAmount)
+        )
+        expect(await ppo.balanceOf(miniSales.address)).to.eq(
+          miniSalesBalanceBefore.sub(saleTokenAmount)
+        )
+      })
+
+      it('reverts if user transfers purchased PPO to blocked recipient', async () => {
```

Comment:
```suggestion
      it('reverts if user transfers purchased PPO to recipient blocked by PPO', async () => {
```

As I mentioned earlier, because our testcases are describing the access level of the recipient, we should be specific as to which contract's access controls.